controller/environment_space_scope_blackbox_test.go

```diff
@@ -0,0 +1,199 @@
+user1	yes		yes			yes
+user2	no		yes			yes
+user3	no		no			no
+*/
+
+var testUser1 = &testauth.Identity{ID: uuid.NewV4(), Email: "user1@test.com", Username: "user1"} // user1
+var testUser2 = &testauth.Identity{ID: uuid.NewV4(), Email: "user2@test.com", Username: "user2"} // user2
+var testUser3 = &testauth.Identity{ID: uuid.NewV4(), Email: "user3@test.com", Username: "user3"} // user3
+
+type EnvironmentSpaceScopeSuite struct {
+	testsuite.DBTestSuite
+	db *gormapp.GormDB
+
+	svc        *goa.Service
+	ctrl       *controller.EnvironmentController
+	authServer *httptest.Server
+
+	ctx1      context.Context
+	ctx2      context.Context
+	ctx3      context.Context
+	spaceID   uuid.UUID // space1
+	publicKey *rsa.PublicKey
+}
+
+func TestEnvironmentSpaceScopeSuite(t *testing.T) {
+	config, err := configuration.New("")
+	require.NoError(t, err)
+	suite.Run(t, &EnvironmentSpaceScopeSuite{DBTestSuite: testsuite.NewDBTestSuite(config)})
+}
+
+func (s *EnvironmentSpaceScopeSuite) SetupSuite() {
+	s.DBTestSuite.SetupSuite()
+
+	s.db = gormapp.NewGormDB(s.DB)
+	s.authServer = s.startAuthServer()
+	authURL := "http://" + s.authServer.Listener.Addr().String() + "/"
+	authService, err := service.NewAuthService(authURL)
+	require.NoError(s.T(), err)
+
+	s.svc = testauth.UnsecuredService("enviroment-test")
+	s.ctrl = controller.NewEnvironmentController(s.svc, s.db, authService)
+	s.spaceID = uuid.NewV4()
+
+	s.ctx1, _, err = testauth.EmbedUserTokenInContext(context.Background(), testUser1)
+	require.NoError(s.T(), err)
+	s.ctx2, _, err = testauth.EmbedUserTokenInContext(context.Background(), testUser2)
+	require.NoError(s.T(), err)
+	s.ctx3, _, err = testauth.EmbedUserTokenInContext(context.Background(), testUser3)
+	require.NoError(s.T(), err)
+
+	tokenMgr, err := token.ReadManagerFromContext(s.ctx1)
+	require.NoError(s.T(), err)
+	keys := tokenMgr.PublicKeys()
+	require.NotNil(s.T(), keys)
+	require.NotEmpty(s.T(), keys)
+	s.publicKey = keys[0]
+}
+
+func (s *EnvironmentSpaceScopeSuite) TearDownSuite() {
+	s.DBTestSuite.TearDownSuite()
+	s.authServer.Close()
+}
+
+func (s *EnvironmentSpaceScopeSuite) TestAllScope() {
+	payload := newCreateEnvironmentPayload("osio-stage", "stage", "cluster1.com")
+	var newEnv *app.EnvironmentSingle
+
+	s.T().Run("user1_create", func(t *testing.T) {
+		_, newEnv = test.CreateEnvironmentCreated(t, s.ctx1, s.svc, s.ctrl, s.spaceID, payload)
+		assert.NotNil(t, newEnv)
+	})
+
+	s.T().Run("user1_list", func(t *testing.T) {
+		require.NotNil(t, newEnv)
+		_, list := test.ListEnvironmentOK(t, s.ctx1, s.svc, s.ctrl, s.spaceID)
+		assert.NotNil(t, list)
+	})
+
+	s.T().Run("user1_show", func(t *testing.T) {
+		require.NotNil(t, newEnv)
+		_, env := test.ShowEnvironmentOK(t, s.ctx1, s.svc, s.ctrl, *newEnv.Data.ID)
+		assert.NotNil(t, env)
+	})
+
+	s.T().Run("user2_create", func(t *testing.T) {
+		require.NotNil(t, newEnv)
+		_, err := test.CreateEnvironmentUnauthorized(t, s.ctx2, s.svc, s.ctrl, s.spaceID, payload)
+		assert.NotNil(t, err)
```

Comment:
you can use `assert.NoError(t, err)` instead (or `require.NoError(...)`

Reply:
err is actually type of `app.JSONAPIErrors` and can't be use with `assert.NoError()` .. it gives below error:
```
controller/environment_space_scope_blackbox_test.go:137:16: cannot use err (type *app.JSONAPIErrors) as type error in argument to assert.Error:
	*app.JSONAPIErrors does not implement error (missing Error method)
```

Reply:
ah my apologies, I read `err` and immediately thought it was an `error` :/